src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -466,4 +475,78 @@ public boolean matches(CtInvocation<?> element) {
 		}).get(0);
 		assertNotNull(invSize);
 	}
+	@Test
+	public void testQueryStepScannWithConsumer() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput" });
+		launcher.addInputResource("./src/test/java/spoon/test/filters/testclasses");
+		launcher.run();
+		
+		class Context {
+			int counter = 0;
+		}
+		Context context = new Context();
+		
+		QueryStep<CtClass<?>> l_qv = launcher.getFactory().getModel().getRootPackage().query().scan(new TypeFilter<>(CtClass.class));
+		
+		assertEquals(0, context.counter);
+		l_qv.forEach(cls->{
+			assertTrue(cls instanceof CtClass);
+			context.counter++;
+		});
+		assertTrue(context.counter>0);
+	}
+	
+	@Test
+	public void testQueryBuilderWithFilterChain() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[] {"--output-type", "nooutput" });
+		launcher.addInputResource("./src/test/java/spoon/test/filters/testclasses");
+		launcher.run();
+		
+		class Context {
+			CtMethod<?> method;
+			int count = 0;
+		}
+		
+		Context context = new Context();
+
+		launcher.getFactory().Package().getRootPackage().query().scan(new TypeFilter<CtMethod<?>>(CtMethod.class))
+		.thenConsume((CtMethod<?> method) -> {context.method = method;})
+		.then(new OverriddenMethodFilter())
```

Comment:
Thanks! I need suggestions and links like that! I do not have theoretic training in these things